tracer/src/Datadog.Trace.ClrProfiler.Managed.Loader/Startup.NetCore.cs

```diff
@@ -108,15 +128,37 @@ private static Assembly ResolveAssembly(string name)
             return null;
         }
 
-        private static bool IsDatadogAssembly(string path, out Assembly cachedAssembly)
+        private static IEnumerable<string> EnumerateProfilerPathEnvironmentVariables()
+        {
+            var archSpecificPath = RuntimeInformation.ProcessArchitecture switch
+                                   {
+                                       Architecture.X64 => "CORECLR_PROFILER_PATH_64",
+                                       Architecture.X86 => "CORECLR_PROFILER_PATH_32",
+                                       Architecture.Arm64 => "CORECLR_PROFILER_PATH_ARM64",
+                                       Architecture.Arm => "CORECLR_PROFILER_PATH_ARM",
+                                       _ => null
+                                   };
+
+            if (!string.IsNullOrEmpty(archSpecificPath))
+            {
+                yield return archSpecificPath!;
+            }
+
+            yield return "CORECLR_PROFILER_PATH";
+        }
+
+        private static bool IsDatadogAssembly(string path, out Assembly? cachedAssembly)
         {
-            for (var i = 0; i < _assemblies.Length; i++)
+            if (_assemblies is not null)
```

Comment:
We can (and probably should) restructure `Startup` so that `_assemblies` is non-nullable, which avoids all these superfluous checks (we _know_ it can't be null here in practice)